tithiwa/group.py

```diff
@@ -142,22 +137,30 @@ def _find_group(self, groupname):
             name = curractive.find_element(By.CSS_SELECTOR, SELECTORS.GROUPS.CONTACTS_SEARCH_NAME).get_attribute(
                 'innerText')
             if name == groupname:
-                group_found = True
+                isfound = True
                 break
             preactive = curractive
-        if not group_found:
-            return None
-        else:
-            return curractive
+        self._wait_for_group_to_open(groupname)
+        return isfound
+
+    def _wait_for_group_to_open(self, groupname):
+        while True:
+            nameofgroup = self._wait_for_an_presence_of_element(SELECTORS.GROUPS.NAME).get_attribute(
+                'innerText')
+            if nameofgroup == groupname:
+                break
+
 
     def _exit_from_group(self):
-        exit_from_group_button = self._wait_for_an_element_to_be_clickable(SELECTORS.GROUPS.EXIT_FROM_GROUP)
```

Comment:
3. Find an efficient way so that we don't need to wait longer. i.e use efficient one of  `something is done` or `something is not done`